articles/a-friendly-beginner-guide-to-typescript/index.md

```diff
@@ -0,0 +1,309 @@
+# What is TypeScript?
+
+TypeScript is a programming language that extends JavaScript by adding types. It saves time catching errors and providing fixes before you run code, so by understanding JavaScript everything comes in handy.
+
+## Table of contents
+
+- TypeScript Basics
+- Compiler Configurations
+- Classes & Interfaces
+- Typescript Features
+- Namespaces and Module
+- Conclusion
+
+## Prerequisite
+
+- Basic knowledge of JavaScript
+
+In this article, I will show you some important guidelines to learn typescript but have in mind that practicing is the best way to learn. Let's get started 😊
+
+## TypeScript Basics
+
+### Boolean
+
+This data type is simply a true/false
+
+```
+let isOpen: Boolean = true;
+```
+
+### Number
+
+TypeScript Supports floating-point values or BigIntegers
+
+```
+let decimal: number = 6;
+let hex: number = 0xf00d;
+let binary: number = 0b1010;
+let octal: number = 0o744;
+let big: bigint = 100n;
+```
+
+### String
+
+TypeScript uses single (') or double(") quote similar to JavaScript.
+
+```
+let colour: string = "yellow";
+color = 'white';
+```
+
+Template Strings is also valid in typeScript and can be used without limitations.
+
+```
+let fullName: string = `Idris Olubisi`;
+let age: number = 47;
+let text: string = `Hello, my name is ${fullName}.`
+```
+
+### Enum
+
+Enum in TypeScript is a special class that contains constant values. By default enum numbering start from the index zero.
+
+```
+enum Fruits {
+  Orange,
+  Apple,
+  Banana,
+}
+// can be accessed by
+let c: Fruits = Fruits.Apple;
+```
+
+You can manually access set the number of an enum.
+
+```
+enum Fruits {
+  Orange = 1,
+  Apple = 4,
+  Banana= 7
+}
+// can be accessed by
+let c: Fruits = Fruits.Bannana;
+```
+
+### Array
+
+TypeScript also supports array similar to javascript and its represented as
+
+```
+let listOfNumbers: number[] = [1, 2, 3, 4, 5, 6];
+```
+
+### Object
+
+An object in TypeScript is an instance that contains a set of key-value pairs. It gives a compilation error whenever you try to access a value that does not exist in a given object.
+
+```
+var person = {
+   firstName:"Idris",
+   lastName:"Olubisi"
+};
+```
+
+### Any
+
+The type `Any` allows us to assign “any” particular value to a variable, similar to what we have in JavaScript.
+
+```
+let amount: any;
+amount = 26;
+amount = true;
+amount = "Hello World";
+amount = [];
+amount = {};
+amount = null;
+amount = undefined;
+```
+
+### Unknown
+
+This works similar to Any but when you try to reassign the value initialized to a new variable then you get an error, Which means anything is assignable to unknown but itself and `any`core types.
```

Comment:
This sentence is hard to read. Please rephrase it.

Reply:
Done